File: app/src/app/core/models/event.ts
/*
 * Este fichero forma parte de UBUMonitorWeb.
 *
 * @author Marcelo Verteramo Pérsico
 */

export type Event = {
  id: number;
  name: string;
  description: string | null;
  format: number;
  courseId: number;
  categoryId: number | null;
  groupId: number | null;
  userId: number | null;
  repeatId: number | null;
  module: string | null;
  instance: number | null;
  type: string;
  startTs: number;
  durationTs: number;
  isVisible: boolean;
  uuid: string | null;
  sequence: number;
  modifiedTs: number;
  subscriptionId: number | null;
};

File: api/src/main/kotlin/es/ubu/lsi/ubumonitorweb/domain/Event.kt
/*
 * Este fichero forma parte de UBUMonitorWeb.
 *
 * @author Marcelo Verteramo Pérsico
 */

package es.ubu.lsi.ubumonitorweb.domain

data class Event(
  val id: Int,
  val name: String,
  val description: String?,
  val format: Int,
  val courseId: Int,
  val categoryId: Int?,
  val groupId: Int?,
  val userId: Int?,
  val repeatId: Int?,
  val module: String?,
  val instance: Int?,
  val type: String,
  val startTs: Long,
  val durationTs: Long,
  val isVisible: Boolean,
  val uuid: String?,
  val sequence: Int,
  val modifiedTs: Long,
  val subscriptionId: Int?,
)
